pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -213,53 +216,11 @@ def work(self):
                                         else:
                                             logger.log(
                                             'Failed to evolve {}!'.format(pokemon_name))
-
-                                    if self.should_release_pokemon(pokemon_name, cp, pokemon_potential, response_dict):
-                                        # Transfering Pokemon
-                                        pokemon_to_transfer = list(
-                                            Set(id_list2) - Set(id_list1))
-                                        if len(pokemon_to_transfer) == 0:
-                                            raise RuntimeError(
-                                                'Trying to transfer 0 pokemons!')
-                                        self.transfer_pokemon(pokemon_to_transfer[0])
-                                        self.bot.metrics.released_pokemon()
-                                        logger.log(
-                                            '{} has been exchanged for candy!'.format(pokemon_name), 'green')
+                                    worker = PokemonTransferWorker(self.bot)
+                                    worker.release_catched_pokemon(pokemon_name, pokemon_to_transfer[0], cp, pokemon_potential)
```

Comment:
This instantiation is wrong in 2 ways:

1. You're transferring the captured pokemon without any verification. 
2. The event loop already has a PokemonTransferWorker.work call, so no need to release pokemon in here.